zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -234,3 +234,60 @@ impl GetBlockTemplate {
         }
     }
 }
+
+/// Error response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ProposalRejectReason {
+    /// Block rejected as invalid
+    Rejected,
+}
+
+/// Response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged, rename_all = "kebab-case")]
+pub enum ProposalResponse {
+    /// Block was not successfully submitted, return error
+    ErrorResponse {
+        /// Reason the proposal was invalid as-is
+        reject_reason: ProposalRejectReason,
+
+        /// The getblocktemplate RPC capabilities supported by Zebra.
+        capabilities: Vec<String>,
+    },
+
+    /// Block successfully proposed, returns null
```

Comment:
This seems to serialize as `"valid"`, not `null`.

To automatically serialize as `null`, use an `Option<ErrorResponse>` instead of the `ProposalResponse` struct.

Reply:
I think it serializes to null (see https://serde.rs/enum-representations.html#untagged), I'll double check.